test/Abp.ZeroCore.Tests/Zero/Notifications/RealTimeNotifier_Tests.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Threading.Tasks;
+using Abp.Notifications;
+using Abp.Runtime.Session;
+using Shouldly;
+using Xunit;
+
+namespace Abp.Zero.Notifications
+{
+    public class RealTimeNotifier_Tests : AbpZeroTestBase
+    {
+        private readonly INotificationPublisher _publisher;
+        private readonly FakeRealTimeNotifier1 _fakeRealTimeNotifier1;
+        private readonly FakeRealTimeNotifier2 _fakeRealTimeNotifier2;
+
+        public RealTimeNotifier_Tests()
+        {
+            _publisher = LocalIocManager.Resolve<INotificationPublisher>();
+            _fakeRealTimeNotifier1 = LocalIocManager.Resolve<FakeRealTimeNotifier1>();
+            _fakeRealTimeNotifier2 = LocalIocManager.Resolve<FakeRealTimeNotifier2>();
+        }
+
+        [Fact]
+        public async Task Should_Notify_Using_Custom_Notifiers()
+        {
+            //Arrange
+            var notificationData = new NotificationData();
+
+            //Act
+            await _publisher.PublishAsync("TestNotification", notificationData, severity: NotificationSeverity.Success, userIds: new[] { AbpSession.ToUserIdentifier() });
+
+            //Assert
+            _fakeRealTimeNotifier1.IsSendCalled.ShouldBeTrue();
```

Comment:
Would it be better to use `NSubstitute`?

E.g.
https://github.com/aspnetboilerplate/aspnetboilerplate/blob/6fd9c77c1fb5f3241b5eab24f293d064fa5d2108/test/Abp.Tests/Notifications/NotificationPublisher_Tests.cs#L35-L43

Reply:
Yes, used NSubstitute in https://github.com/aspnetboilerplate/aspnetboilerplate/pull/4330/commits/bbba9ba4110ff93cc0d34c0b38fe9b4561fa4086.